### DISCUSSION TITLE
feat: call function in ionic from capacitor IOS while app in background without opening app

### DISCUSSION THREAD

**Discussion by MarwaAbuEssa** [ionic-team/capacitor#5475] [Q&A] [answered]
## Feature Request
call function from capacitor IOS while app in background without opening app
### Description

HI ,

I'm implementing Voip push notification from capacitor IOS project and it works perfect .my issue is I need to send to ionic/angular app about the incoming call if answered / declined while ionic app is closed (in background) without opening the ionic/angular app.

for example I'm trying to do the following scenario :

1- Voip notification sent to capacitor IOS app 
2- meanwhile I need the app to connect to voip server ( which is already implemented and working from ionic/angular side and working perfect while app opened)
3- if user answered from CallKit and opened the app ( will open the ionic/angular app ) will display the appropriate interface about running call  

### Platform(s)

ionic 5
angular 10
capacitor 3.34
swift 5.1

### Preferred Solution

my question => Is there a capability to send data and call functions in ionic via capacitor while IOS app is closed ?

### Additional Context

Regards,
Marwa

**Comment by jcesarmobile** [comment]
Converting the issue to discussion since this is a question.

In general, when the app goes into background the WebView stops executing javascript, so you can't do any native to JS communication if that happens.
I personally have not tried with VOIP notifications, so can't tell for sure. 

About how to send data or execute javascript from native, you can check app plugin, it uses notification center for listening for system and custom notifications 
https://github.com/ionic-team/capacitor-plugins/blob/main/app/ios/Plugin/AppPlugin.swift#L8-L22

**Reply by MarwaAbuEssa** [reply]
Thank you so much for your support . let's keep it for discussion if anyone have idea about it.

**Comment by MarwaAbuEssa** [comment, marked as answer]
the issue is solved , in IOS the Voip push notification wakes up the app in background and I can see the app component code is working . Thank you

**Reply by illiaravkis** [reply]
Hello @MarwaAbuEssa 
I hope to know about your solution
